test/unit_tests/test_package_version.py

```diff
@@ -40,3 +41,30 @@ def test_update_versions(self):
                 self.assertEqual('<package><version>7.6.5</version></package>', contents)
         finally:
             shutil.rmtree(root_dir)
+
+    def test_update_changelog_unicode(self):
+        '''Test that updating the changelog does not
+        throw an exception on unicode characters.
+        '''
+        with tempfile.NamedTemporaryFile() as temp_file:
+            missing_changelogs_but_forthcoming = {}
+            # Mock the Changelog object from catkin_pkg
+            mock_changelog = mock.Mock()
+            # Create a changelog entry with a unicode char.
+            mock_changelog.rst = ('^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n'
+                                  'Changelog for package fake_pkg\n'
+                                  '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n'
+                                  '\n'
+                                  'Forthcoming\n'
+                                  '-----------\n'
+                                  '* This is my changelog entry\n'
+                                  '* This is a line that has unicode' u'\xfc''\n'
+                                  '\n'
+                                  '0.0.9 (2017-01-30)\n'
+                                  '------------------\n'
+                                  '* This is old version.\n')
+
+            # Create tuple with expected entires.
+            missing_changelogs_but_forthcoming['fake_pkg'] = (temp_file.name, mock_changelog, 'Forthcoming')
+            # Should not raise an exception
+            update_changelog_sections(missing_changelogs_but_forthcoming, '1.0.0')
```

Comment:
Worthwhile to actually validate the output since you're here anyway?

Either way, LGTM.

Reply:
I can, but the issue here would be adding a sleep to the test.

I've tried flushing the file in the past and calling the system flush, but still end up with a race condition where the file hasn't finished populating to disk yet.

Reply:
Doesn't seem to be effected by disk buffer issues on my PC, added a read-back assertion.

Reply:
Readback test added, with python2 & 3 working.